physical/aerospike/aerospike_test.go

```diff
@@ -0,0 +1,88 @@
+package aerospike
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	aero "github.com/aerospike/aerospike-client-go"
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/testhelpers/docker"
+	"github.com/hashicorp/vault/sdk/helper/logging"
+	"github.com/hashicorp/vault/sdk/physical"
+)
+
+func TestAerospikeBackend(t *testing.T) {
+	cleanup, config := prepareAerospikeContainer(t)
+	defer cleanup()
+
+	logger := logging.NewVaultLogger(log.Debug)
+
+	b, err := NewAerospikeBackend(map[string]string{
+		"hostname":  config.hostname,
+		"port":      config.port,
+		"namespace": config.namespace,
+		"set":       config.set,
+	}, logger)
+
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	physical.ExerciseBackend(t, b)
+	physical.ExerciseBackend_ListPrefix(t, b)
+}
+
+type aerospikeConfig struct {
+	hostname  string
+	port      string
+	namespace string
+	set       string
+}
+
+func prepareAerospikeContainer(t *testing.T) (func(), *aerospikeConfig) {
+	runner, err := docker.NewServiceRunner(docker.RunOptions{
+		ImageRepo:       "aerospike/aerospike-server",
+		ContainerName:   "aerospikedb",
+		ImageTag:        "latest",
+		Ports:           []string{"3000/tcp", "3001/tcp", "3002/tcp", "3003/tcp"},
+		DoNotAutoRemove: true,
```

Comment:
Is this strictly necessary? This leaves dangling containers on the host, and may result in errors if tests are ran multiple times due to colliding container name.